std/assembly/internal/date.ts

```diff
@@ -0,0 +1,381 @@
+export class CalendarDate {
+  year: i32;
+  month: i32; // 1 ~ 12
+  dayOfMonth: i32; // 1 ~ 31
+  dayOfWeek: i32; // 1 ~ 7
+  hours: i32;
+  minutes: i32;
+  seconds: i32;
+  milliseconds: i32;
+  timeOfDay: i32;
+  leapYear: bool;
+  isNormalized: bool;
+
+  // The represents date millis
+  millis: i64;
+
+  constructor(millis: i64) {
+    this.millis = millis;
+    this.isNormalized = false;
+  }
+}
+
+class CalendarUtils {
+
+  static SECOND_IN_MILLIS: i32 = 1000;
+  static MINUTE_IN_MILLIS: i32 = 60000; // 1000 * 60
+  static HOUR_IN_MILLIS: i32 = 3600000; // 1000 * 60 * 60
+  static DAY_IN_MILLIS: i32 = 86400000; // 1000 * 60 * 60 * 24
+  static DAY_IN_HOURS: i32 = 24;
+  static YEAR_IN_MONTHS: i32 = 12;
+
+  // The number of days between January 1, 1 and January 1, 1970 (Gregorian)
+  static EPOCH_OFFSET: i32 = 719163;
+
+  static floorDivide(n: i32, d: i32): i32 {
+    return ((n >= 0) ? (n / d) : (((n + 1) / d) - 1));
+  }
+
+  static mod(x: i32, y: i32): i32 {
+    return (x - y * CalendarUtils.floorDivide(x, y));
+  }
+
+  static isLeapYear(gregorianYear: i32): bool {
+    return  (((gregorianYear % 4) == 0)
```

Comment:
I think also need add this check before other `gregorianYear > 0` && ...

Reply:
What about speed up this little bit?
```ts
static isLeapYear(year: i32): bool {
  return (year & 3) == 0 && (((year % 25) | 0) != 0 || (year & 15) == 0);
}
```

Also not sure about additional check  `gregorianYear > 0`. Need more investigate

Reply:
The implement to judge the leap year is cool.